app/src/main/java/org/mozilla/tv/firefox/FirefoxApplication.kt

```diff
@@ -50,6 +51,16 @@ open class FirefoxApplication : LocaleAwareApplication() {
             // Enable crash reporting. Don't add anything above here because if it crashes, we won't know.
             SentryIntegration.init(this, serviceLocator.settingsRepo)
 
+            // Initialize Glean telemetry and hook it to the collection enabled flag
+            Glean.initialize(applicationContext)
+            with(serviceLocator.settingsRepo) {
+                dataCollectionEnabled.observeForever { collectionEnabled ->
+                    if (collectionEnabled != null) {
+                        Glean.setUploadEnabled(collectionEnabled)
```

Comment:
This should be called *before* `Glean.initialize`, otherwise we risk not sending pings at startup, if needed. Or, worse, sending pings if the flag is flipped to off :)

Reply:
Yep, I forgot about that.